x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authz/AuthorizationService.java

```diff
@@ -191,51 +201,170 @@ public void authorize(final Authentication authentication, final String action,
+            // We want to treat shard level actions (those that append '[s]' and/or '[p]' & '[r]') as children
+            return false;
+        }
+        final IndicesRequest indicesRequest;
+        if (request instanceof IndicesRequest) {
+            indicesRequest = (IndicesRequest) request;
+        } else {
+            // Can only handle indices request here
+            return false;
+        }
+
+        final String[] indices = indicesRequest.indices();
+        if (indices == null || indices.length == 0) {
+            // No indices to check
+            return false;
+        }
+
+        final IndicesAccessControl indicesAccessControl = threadContext.getTransient(AuthorizationServiceField.INDICES_PERMISSIONS_KEY);
+        if (indicesAccessControl == null) {
+            // This is almost certainly an error, but leave it to be handled later
+            return false;
+        }
 
-            // sometimes a request might be wrapped within another, which is the case for proxied
-            // requests and concrete shard requests
-            final TransportRequest unwrappedRequest = maybeUnwrapRequest(authentication, originalRequest, action, auditId);
-
-            // Check operator privileges
-            // TODO: audit?
-            final ElasticsearchSecurityException operatorException =
-                operatorPrivilegesService.check(action, originalRequest, threadContext);
-            if (operatorException != null) {
-                listener.onFailure(denialException(authentication, action, originalRequest, operatorException));
-                return;
+        if (Arrays.stream(indices).allMatch(idx -> {
+            if (Regex.isSimpleMatchPattern(idx)) {
+                // The request contains a wildcard
+                return false;
             }
-            operatorPrivilegesService.maybeInterceptRequest(threadContext, originalRequest);
+            IndicesAccessControl.IndexAccessControl iac = indicesAccessControl.getIndexPermissions(idx);
+            return iac != null && iac.isGranted();
+        })) {
+            logger.trace("Short circuit authorization child action [{}] (child of [{}]) on indices [{}]", action, originalAction, indices);
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private void authorizeChildAction(RequestInfo requestInfo, String auditId, ActionListener<Void> listener) {
+        AuthorizationInfo authorizationInfo = threadContext.getTransient(AUTHORIZATION_INFO_KEY);
+        if (authorizationInfo == null) {
+            listener.onFailure(
+                internalError(
+                    "Attempt to authorize child action ["
+                        + requestInfo.getAction()
+                        + "] for ["
+                        + requestInfo.getAuthentication().getUser().principal()
+                        + "] without existing authorization info"
+                )
+            );
```

Comment:
```suggestion
            );
            return;
```

Reply:
Thanks. It was originally a throw, but then I switched it and missed the return.